### DISCUSSION TITLE
Noob question about mergerfs.dedup

### DISCUSSION THREAD

**Discussion by dma9** [trapexit/mergerfs-tools#130] [Q&A]
Me again, with another dumb question, but this time, about mergerfs.dedup. I tentatively tried the following command:

`mergerfs.dedup -i diff-hash -d newest -vvv /vdata`

Where `/vdata` is the mountpoint of my drive array consisting of 16 drives each mounted to `/mnt/data/dataX`. I guess where I'm having a bit of trouble is understanding the output of the command. Here's one entry from the long list that the command generates as an example:

```
# /vdata/media/short/Uncle Bob/poster.jpg
#  1: /mnt/data/data11/media/short/Uncle Bob/poster.jpg
#   - uid:  2000; gid:  2000; mode: 100777; size: 178.4KB; mtime: 1546835003.511199
rm -vf '/mnt/data/data11/media/short/Uncle Bob/poster.jpg'
```
The first line seems to identify a file on the mergerfs mountpoint that presumably is duplicated - is that correct? Then the second and third lines seem to identify that same file's location on one of the underlying drives comprising the mergerfs array - is that correct? Then the last line seems to indicate which file mergerfs.dedup will delete which seems to be the same file identified in lines 2 and 3.

I guess the source of my confusion is that it doesn't seem to show multiple files and their locations on the underlying drives. It only shows, I think, the one that will be deleted. Is that right? I was just worried that the first line shows the file that would be kept and mergerfs.dedup is mistakenly then showing the only underlying file on the underlying drive as a duplicate, when there actually isn't a duplicate.

Perhaps along similar lines, would it be possible to force mergerfs.dedup to show information on all of the multiple copies it has found? So for example in relation to the entry above, assuming there is actually a duplicate on, say,  /mnt/data/data8, something like the following: 

```
# /vdata/media/short/Uncle Bob/poster.jpg
#  1: /mnt/data/data11/media/short/Uncle Bob/poster.jpg
#   - uid:  2000; gid:  2000; mode: 100777; size: 178.4KB; mtime: 1546835003.511199
#  2: /mnt/data/data8/media/short/Uncle Bob/poster.jpg
#   - uid:  2000; gid:  2000; mode: 100777; size: 178.4KB; mtime: 1546835003.511200
rm -vf '/mnt/data/data11/media/short/Uncle Bob/poster.jpg'
```

But perhaps I'm just misunderstanding the output? Anyway, any guidance on the above would be most appreciated.

**Comment by trapexit** [comment]
>The first line seems to identify a file on the mergerfs mountpoint that presumably is duplicated - is that correct? Then the second and third lines seem to identify that same file's location on one of the underlying drives comprising the mergerfs array - is that correct? Then the last line seems to indicate which file mergerfs.dedup will delete which seems to be the same file identified in lines 2 and 3.

Yes. Yes. No, not exactly. 

The program generates a shell script that can be run to do the job. Unless --execute is given in which case it acts on it. '#' is a comment. So yes... it's showing you details about what it found but the last line is literally the command "rm -vr <file" which would remove the file if you saved the script and executed it. "#" indicates a comment so it would be skipped in the script.

It removes from the list the one it isn't going to remove when using automatic removal like you set because it doesn't want to suggest it's removing things it isn't. https://github.com/trapexit/mergerfs-tools/blob/master/src/mergerfs.dedup#L384
